dask/dataframe/core.py

```diff
@@ -2490,9 +2490,14 @@ def __getitem__(self, key):
         raise NotImplementedError(key)
 
     def __setitem__(self, key, value):
-        if isinstance(key, (tuple, list)):
+        if isinstance(key, (tuple, list)) and isinstance(value, DataFrame):
             df = self.assign(**{k: value[c]
                                 for k, c in zip(key, value.columns)})
+
+        elif isinstance(key, pd.Index) and not isinstance(value, DataFrame):
+            key = list(key)
+            df = self.assign(**{k: value
+                                for k in key})
```

Comment:
Style nit: these can probably be placed on the same line.  We often split things when they are long enough that reading them is awkward.  This one seems quite short though.

Reply:
This is fixed too.